galsim/bandpass.py

```diff
@@ -413,14 +413,37 @@ def truncate(self, blue_limit=None, red_limit=None, relative_throughput=None):
         This function does not remove any intermediate wavelength ranges, but see thin() for
         a method that can thin out the intermediate values.
 
-        @param blue_limit       Truncate blue side of bandpass here. [default: None]
-        @param red_limit        Truncate red side of bandpass here. [default: None]
+        When truncating a bandpass that already has an assigned zeropoint, there are several
+        possibilities for what should happen to the new (returned) bandpass by default.  If red
+        and/or blue limits are given, then the new bandpass will have no assigned zeropoint because
+        it is difficult to predict what should happen if the bandpass is being arbitrarily
+        truncated.  If `relative_throughput` is given, often corresponding to low-level truncation
+        that results in little change in observed quantities, then the new bandpass is assigned the
+        same zeropoint as the original.  This default behavior is called 'auto'.  The user can also
+        give boolean True or False values.
+
+        @param blue_limit           Truncate blue side of bandpass here. [default: None]
+        @param red_limit            Truncate red side of bandpass here. [default: None]
         @param relative_throughput  Truncate leading or trailing wavelengths that are below
-                                this relative throughput level.  (See above for details.)
-                                [default: None]
+                                    this relative throughput level.  (See above for details.)
+                                    [default: None]
+        @param preserve_zp          If True, the new truncated Bandpass will be assigned the same
+                                    zeropoint as the original.  If False, the new truncated Bandpass
+                                    will have a zeropoint of None. If 'auto', the new truncated
+                                    Bandpass will have the same zeropoint as the original when
+                                    truncating using `relative_throughput`, but will have a
+                                    zeropoint of None when truncating using 'blue_limit' and/or
+                                   'red_limit'.  [default: 'auto']
 
         @returns the truncated Bandpass.
         """
+        if preserve_zp == 'auto':
+            if relative_throughput is not None: preserve_zp = True
```

Comment:
I think it's probably least confusing if these are made mutually exclusive.  Also, we should guard against setting a `blue_limit` that's bluer than the original blue_limit and similarly for red_limit.